sandbox/cgroup/cgroup_test.go

```diff
@@ -251,30 +251,38 @@ func (s *cgroupSuite) TestProcessPathInTrackingCgroup(c *C) {
 	defer dirs.SetRootDir(dirs.GlobalRootDir)
 	dirs.SetRootDir(d)
 
-	restore := cgroup.MockVersion(cgroup.V2, nil)
-	defer restore()
-
 	f := filepath.Join(d, "proc", "1234", "cgroup")
 	c.Assert(os.MkdirAll(filepath.Dir(f), 0755), IsNil)
 
-	for _, scenario := range []struct{ cgroups, path, errMsg string }{
-		{cgroups: "", path: "", errMsg: "cannot find tracking cgroup"},
-		{cgroups: noise + "", path: "", errMsg: "cannot find tracking cgroup"},
-		{cgroups: noise + "0::/foo", path: "/foo"},
-		{cgroups: noise + "1:name=systemd:/bar", path: "/bar"},
+	for _, scenario := range []struct {
+		cgVersion             int
+		cgroups, path, errMsg string
+	}{
+		{cgVersion: cgroup.V2, cgroups: "", path: "", errMsg: "cannot find tracking cgroup"},
+		{cgVersion: cgroup.V2, cgroups: noise + "", path: "", errMsg: "cannot find tracking cgroup"},
+		{cgVersion: cgroup.V2, cgroups: noise + "0::/foo", path: "/foo"},
+		{cgVersion: cgroup.V2, cgroups: noise + "1:name=systemd:/bar", errMsg: "cannot find tracking cgroup"},
+		// In only V1 is mounted, then the same configuration works
+		{cgVersion: cgroup.V1, cgroups: noise + "1:name=systemd:/bar", path: "/bar"},
 		// First match wins (normally they are in sync).
-		{cgroups: noise + "1:name=systemd:/bar\n0::/foo", path: "/bar"},
-		{cgroups: "0::/tricky:path", path: "/tricky:path"},
-		{cgroups: "1:ctrl" /* no path */, errMsg: `cannot parse proc cgroup entry ".*": expected three fields`},
-		{cgroups: "potato:foo:/bar" /* bad ID number */, errMsg: `cannot parse proc cgroup entry ".*": cannot parse cgroup id "potato"`},
+		{cgVersion: cgroup.V1, cgroups: noise + "1:name=systemd:/bar\n0::/foo", path: "/bar"},
+		{cgVersion: cgroup.V2, cgroups: noise + "1:name=systemd:/bar\n0::/foo", path: "/foo"},
+		{cgVersion: cgroup.V2, cgroups: "0::/tricky:path", path: "/tricky:path"},
+		{cgVersion: cgroup.V2, cgroups: "1:ctrl" /* no path */, errMsg: `cannot parse proc cgroup entry ".*": expected three fields`},
+		{cgVersion: cgroup.V2, cgroups: "potato:foo:/bar" /* bad ID number */, errMsg: `cannot parse proc cgroup entry ".*": cannot parse cgroup id "potato"`},
 	} {
+		restoreCGVersion := cgroup.MockVersion(scenario.cgVersion, nil)
+
```

Comment:
(nitpick^2) you could `defer restore()` here which is a bit wasteful but slightly more idiomatic